app-sdk/src/lib.rs

```diff
@@ -125,3 +125,29 @@ macro_rules! bootstrap {
         }
     };
 }
+
+#[no_mangle]
+#[inline(never)]
+#[cfg(target_arch = "riscv32")]
+pub unsafe extern "C" fn memcpy(dest: *mut u8, src: *const u8, n: usize) -> *mut u8 {
+    Ecall::sys_memcpy(dest, src, n)
+}
+
+#[no_mangle]
+#[inline(never)]
```

Comment:
Why `inline(never)`? To me it seems like we should somehow override the LLVM intrinsic instead.

Reply:
Yeah, probably not needed; I was worried that inline might for some reason interfere with `#[no_mangle]`, but that seems unfounded.